activerecord/test/cases/destroy_association_later_test.rb

```diff
@@ -0,0 +1,263 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+
+setup_destroy_later do
+  require "models/book_destroy_later"
+  require "models/essay_destroy_later"
+  require "models/tag"
+  require "models/tagging"
+  require "models/essay"
+  require "models/category"
+  require "models/post"
+  require "models/content"
+  require "models/destroy_later_parent"
+  require "models/destroy_later_parent_soft_delete"
+  require "models/dl_keyed_belongs_to"
+  require "models/dl_keyed_belongs_to_soft_delete"
+  require "models/dl_keyed_has_one"
+  require "models/dl_keyed_join"
+  require "models/dl_keyed_has_many"
+  require "models/dl_keyed_has_many_through"
+end
+
+class DestroyAssociationLaterTest < ActiveRecord::TestCase
+  include ActiveJob::TestHelper
+
+  test "destroying a book destroy later enqueues the has_many through tags to be deleted" do
+    setup_destroy_later do
+      tag = Tag.create!(name: "Der be treasure")
+      tag2 = Tag.create!(name: "Der be rum")
+      book = BookDestroyLater.create!
+      book.tags << [tag, tag2]
+      book.save!
+      book.destroy
+
+      assert_difference -> { Tag.count }, -2 do
+        perform_enqueued_jobs only: ActiveRecord::DestroyAssociationLaterJob
+      end
+    end
+  end
+
+  test "destroying a a scoped has_many through only deletes within the scope deleted" do
+    setup_destroy_later do
+      tag = Tag.create!(name: "Der be treasure")
+      tag2 = Tag.create!(name: "Der be rum")
+      parent = DestroyLaterParent.create!
+      parent.tags << [tag, tag2]
+      parent.save!
+
+      parent2 = DestroyLaterParent.find(parent.id)
+      parent2.destroy
+
+      assert_difference -> { Tag.count }, -1 do
+        perform_enqueued_jobs only: ActiveRecord::DestroyAssociationLaterJob
+      end
+      assert_raises ActiveRecord::RecordNotFound do
+        tag2.reload
+      end
+      assert tag.reload
+    end
+  end
+
+
+  test "enqueues the has_many through to be deleted with custom primary key" do
```

Comment:
What is this test trying to assert? If it works when the `has_many` association has a primary_key option or when the model that defines association has a custom primary key?

I'm trying to reduce the number of new tables added and I was trying to reuse some of the existing models but I'm having a hard time to know what exactly is being tested here.